### PR TITLE
Code quality improvements

### PR DESCRIPTION
**Pull request by GhaziTriki**
## Type

- Critical bugfix
- Non critical bugfix
- Enhancement

## Resolves the following issues

Improve general code quality.

## Pull request description

The list of improvements is on each of the commits comments.

### CODE REVIEW

File: src/Api/V1/Engine/Api.php [hunk 1]
```diff
@@ -163,12 +163,12 @@ public function initialize()
...
-                if (isset($parameters['debug']) && $parameters['debug'] == 'false') {
-                    // do nothing
-                } else {
-                    throw $e;
-                }
+            if (
+                $this->getContainer()->getParameter('kernel.debug')
+                &&
+                (!isset($parameters['debug']) || $parameters['debug'] === 'true')
+            ) {
```
Comment (carakas) [inline src/Api/V1/Engine/Api.php]: I am personally more a fan of 
```
 if ($this->getContainer()->getParameter('kernel.debug')
    && (!isset($parameters['debug']) || $parameters['debug'] === 'true')) {
```
but that might just be taste

File: src/Backend/Core/Engine/Cronjob.php [hunk 2]
```diff
@@ -175,7 +181,7 @@ public function loadConfig()
         $this->config = new $configClass($this->getKernel(), $this->getModule());
 
         // set action
-        $action = ($this->config->getDefaultAction() !== null) ? $this->config->getDefaultAction() : 'Index';
+        $this->action = ($this->config->getDefaultAction() !== null) ? $this->config->getDefaultAction() : 'Index';
```
Comment (carakas) [inline src/Backend/Core/Engine/Cronjob.php]: did you test a cronjob after this change?

**Comment by carakas** [comment]
👍

File: src/Backend/Core/Engine/Cronjob.php (hunk 2 above)
Comment (GhaziTriki) [inline src/Backend/Core/Engine/Cronjob.php]: It breaks cronjob, I reverted it back and should be working again.

**Pull request merged**